Explorer/Assets/DCL/SceneLoadingScreens/SceneLoadingScreenController.cs

```diff
@@ -0,0 +1,164 @@
+        }
+
+        protected override void OnViewClose()
+        {
+            base.OnViewClose();
+
+            tipsRotationCancellationToken?.SafeCancelAndDispose();
+        }
+
+        protected override UniTask WaitForCloseIntent(CancellationToken ct)
+        {
+            return ShowTipsAsync(ct)
+                  .ContinueWith(() =>
+                   {
+                       tipsRotationCancellationToken = tipsRotationCancellationToken.SafeRestart();
+                       RotateTipsOverTime(tips.Duration, tipsRotationCancellationToken.Token).Forget();
+                   })
+                  .ContinueWith(() => WaitUntilWorldIsLoadedAsync(ct));
+        }
+
+        private async UniTask ShowTipsAsync(CancellationToken ct)
+        {
+            tips = await sceneTipsProvider.Get(ct);
+
+            List<SceneTips.Tip> list = ListPool<SceneTips.Tip>.Get();
+
+            if (tips.Random)
+                tips.Tips.Shuffle(list);
+            else
+                list.AddRange(tips.Tips);
+
+            foreach (SceneTips.Tip tip in list)
+                viewInstance.AddTip(tip);
+
+            currentTip = 0;
+            viewInstance.ShowTip(currentTip);
+
+            ListPool<SceneTips.Tip>.Release(list);
+        }
+
+        private async UniTask WaitUntilWorldIsLoadedAsync(CancellationToken ct)
+        {
+            async UniTaskVoid UpdateProgressBarAsync(CancellationToken ct)
+            {
+                do
+                {
+                    try
+                    {
+                        float progress = Mathf.Clamp01(await inputData.AsyncLoadProcessReport.ProgressCounter.WaitAsync(ct));
+                        await UniTask.SwitchToMainThread(ct);
+                        viewInstance.ProgressBar.normalizedValue = progress;
+                        progressLabel!.Value = (int)(progress * 100);
+                    }
+                    catch (OperationCanceledException) { }
+                }
+                while (viewInstance.ProgressBar.normalizedValue < 1);
+            }
+
+            var progressUpdatingCancellationToken = CancellationTokenSource.CreateLinkedTokenSource(ct);
+            UpdateProgressBarAsync(progressUpdatingCancellationToken.Token).Forget();
+
+            try
+            {
+                await inputData.AsyncLoadProcessReport.CompletionSource.Task;
+                progressUpdatingCancellationToken.Cancel();
+                ct.ThrowIfCancellationRequested();
+                viewInstance.ProgressBar.normalizedValue = 1f;
+                progressLabel!.Value = 100;
+            }
+            catch { }
+        }
+
+        private void ShowTip(int index)
+        {
+            if (index < 0)
+                index = tips.Tips.Count - 1;
+
+            index %= tips.Tips.Count;
+
+            viewInstance.ShowTip(index);
+
+            currentTip = index;
+        }
+
+        private async UniTaskVoid RotateTipsOverTime(TimeSpan frequency, CancellationToken ct)
```

Comment:
Should be a `UniTask` - a part of `UniTask.WhenAll`

Reply:
Is also invoked when clicking next/prev tip buttons, and is a task that will never finish as mentioned earlier